app/controllers/avo/base_controller.rb

```diff
@@ -68,11 +68,6 @@ def index
         @resource.dup.hydrate(record: record)
       end
 
-      # Temporary fix for visible blocks when geting fields for header
-      # Hydrating with last record so resource.record != nil
-      # This is keeping same behavior from <= 3.4.1
-      @resource.hydrate(record: @records.last)
-
       set_component_for __method__
```

Comment:
Fixes https://github.com/avo-hq/avo/issues/2653#issuecomment-2044518084

Reply:
I understand this, but didn't we used to have a problem with checking the the presence of `record` in blocks, or this is a different fix?

Reply:
> didn't we used to have a problem with checking the the presence of `record` in blocks

## TL;DR; 

With new computed header logic this is no longer an issue and this bug can be safely removed fixing many edge cases.

## Long version

When we discovered this bug we removed hydration with the last record on the resource which resulted in `resource.record` and `record` being `nil` as expected on the Index when fetching fields for the header. This rollback of hydration was done solely to prevent breaking the provided example.

For example:

```ruby
field :name, as: :text, visible: -> { record.something? } 
```

When checking fields for header this visible block was getting executed and when we removed the last record hydration it was breaking since record was `nil`. 

With this PR we can safely remove this wrong hydration since header now is calculated having in consideration each row fields. This hydration was causing so many other issues, and many of them not "breakable" but weird edge cases.

Through the approach in this PR, we now execute `visible: -> { record.something? }` for each row, ensuring that record is hydrated, and then compute the header based on the visible fields of each row.